src/shogun/classifier/Perceptron.cpp

```diff
@@ -77,6 +78,10 @@ void CPerceptron::iteration()
 			v.add(gradient, w);
 		}
 	}
+
+	observe(ObservedValue::make_observation(m_current_iteration, "w", "Model weights", m_w));
+	observe(ObservedValue::make_observation(m_current_iteration, "bias", "Model bias", bias));
```

Comment:
Unfortunately, this is not possible since otherwise, we would need to have a templated `observe` which brings back all the usual problem we are facing with `put()`.

Reply:
now that the put is "solved", is it reasonable to ask about this again? :)

Reply:
For sure ;) now everything is possible

Reply:
haha, so yeah I think it would be a bit neater.
One thing we need to keep in mind is that if we make `observe` templated, then it cannot be virtual anymore so we cannot really overload it

Reply:
Ideally, `observe` does not need to be overloaded.

Reply:
ok, then why not give that a try?